Makefile

```diff
@@ -1,4 +1,7 @@
 ############################# Main targets #############################
+# Install all tools and builds binaries.
+install: update-tools bins
```

Comment:
very nice.

this looks good, but this also made me wonder about a couple of things (i don't mean this as a change request, but as thinking outloud / wondering what you think about this -- we can move the discussion someplace other than this PR):
* i wonder if we should just make (hehe) this point to an explicit target, e. g. `make install`, to make docker build a bit more explicit and less fragile: https://github.com/temporalio/temporal/blob/eb41b59071e7ea0d86674f7e0d86f3c256479c72/Dockerfile#L24
* by convention (https://www.gnu.org/prep/standards/html_node/Standard-Targets.html#Standard-Targets), `make install` seems to imply that the application is going to get "installed". I am not super-clear what that would mean in our context, ;) but i wonder if we want to simply retire the `install` target, and either give `update-tools` + `bins` an alias that has less meaning attached to it, or just invoke the two targets in our Dockerfile directly.

Reply:
Changing `Dockerfile` to explicit target name makes sense for me. It should be `make bins` though, because we don't need any tools to build server now (all proto generated files are checked in).
As for `install` target. My mental model is that `install` installs everything for development and running service. Once you run `make install` you should be able to develop and run it. So it perfectly fits it.